### PR TITLE
feat(pci.kubernetes): refacto oidc provider 

### PR DESCRIPTION
**Pull request by Eric-ciccotti**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix TAPC-2088
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Refacto of oidc provider form

### CODE REVIEW

File: packages/manager/apps/pci-kubernetes/src/api/data/kubernetes.ts [hunk 1]
```diff
@@ -81,13 +81,13 @@ export const getClusterRestrictions = async (
...
-  signingAlgorithms: string | null;
-  caContent: string;
-  requiredClaim: string | null;
+  usernameClaim?: string;
+  usernamePrefix?: string;
+  groupsClaim?: string | null;
+  groupsPrefix?: string;
+  signingAlgorithms?: string | null;
+  caContent?: string;
+  requiredClaim?: string | null;
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-kubernetes/src/api/data/kubernetes.ts:90]: string undefined or null it seems like a lot isn't it ?

File: packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+  useAddOrUpdateOidcProvider,
+} from '@/api/hooks/useKubernetes';
+import { KUBE_TRACK_PREFIX } from '@/tracking.constants';
+
+const oidcSchema = z.object({
+  issuerUrl: z
+    .string()
+    .url({ message: 'Issuer URL must be a valid URL' })
+    .refine((url) => url.startsWith('https://'), {
+      message: 'Issuer URL must use HTTPS protocol',
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx:39]: this needs translation ?

File: packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+    }),
+  clientId: z
+    .string()
+    .min(1, { message: 'common:common_field_error_required' }),
```
Comment (ppprevost) [inline packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx:43]: exemple of translate

File: packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+                )}
+              </OsdsText>
+
+              <OsdsText
+                size={ODS_TEXT_SIZE._400}
+                level={ODS_TEXT_LEVEL.body}
+                color={ODS_TEXT_COLOR_INTENT.text}
+              >
+                {t(
+                  'pci_projects_project_kubernetes_details_service_add_oidc_provider_field_url_extra',
```
Comment (ppprevost) [inline packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx:219]: add only ?

File: packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+          </div>
+        ) : (
+          <form onSubmit={handleSubmit(onSubmit)} className="mt-6">
+            <OsdsText
+              size={ODS_TEXT_SIZE._400}
+              level={ODS_TEXT_LEVEL.body}
+              color={ODS_THEME_COLOR_INTENT.text}
+            >
+              {t(
+                'pci_projects_project_kubernetes_details_service_add_oidc_provider_description',
```
Comment (ppprevost) [inline packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx:190]: add only ?

File: packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+                  }_oidc_provider_field_client_id`,
+                )}
+              </OsdsText>
+              <OsdsText
+                size={ODS_TEXT_SIZE._400}
+                level={ODS_TEXT_LEVEL.body}
+                color={ODS_TEXT_COLOR_INTENT.text}
+              >
+                {t(
+                  'pci_projects_project_kubernetes_details_service_add_oidc_provider_field_client_id_extra',
```
Comment (ppprevost) [inline packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx:262]: idem

File: packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+          ns={`${isUpdate ? 'update-oidc-provider' : 'add-oidc-provider'}`}
+        >
+          {(_t) =>
+            _t(
+              `pci_projects_project_kubernetes_details_service_${
+                isUpdate ? 'update' : 'add'
+              }_oidc_provider_request_error`,
+              {
+                message:
+                  error?.response?.data?.message || error?.message || null,
```
Comment (ppprevost) [inline packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx:123]: error is defined

File: packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+    .optional(),
+  caContent: z.string().optional(),
+  requiredClaim: z
+    .string()
+    .nullable()
+    .optional(),
+});
+
+export type OidcFormValues = z.infer<typeof oidcSchema> &
+  Required<{ issuerUrl: string; clientId: string }>;
```
Comment (ppprevost) [inline packages/manager/apps/pci-kubernetes/src/pages/oidc-provider/AddOrUpdateOIDCProvider.page.tsx:63]: way ==> Omit<d, 'issuerUrl'> & { issuerUrl: string };

File: packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.ts [hunk 9]
```diff
@@ -33,6 +32,7 @@ import {
 import { getPrivateNetworkName } from '../data/network';
 import { useAllPrivateNetworks } from './useNetwork';
 import { mapPluginsFromArrayToObject } from '../data/plugins';
+import { OidcFormValues } from '@/pages/oidc-provider/AddOrUpdateOIDCProvider.page';
```
Comment (anooparveti) [inline packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.ts:35]: **nitpick:** if it needs to be reused, define it in "types" folder and use it. `pages` is not the correct folder to have reusable types.

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/pci-kubernetes/src/routes.tsx [hunk 10]
```diff
@@ -125,13 +125,13 @@ export default [
               {
                 path: 'add-oidc-provider',
                 ...lazyRouteConfig(() =>
-                  import('@/pages/oidc-provider/AddOIDCProvider.page'),
+                  import('@/pages/oidc-provider/AddOrUpdateOIDCProvider.page'),
```
Comment (anooparveti) [inline packages/manager/apps/pci-kubernetes/src/routes.tsx:128]: **nitpick:** Each page/route must be a folder according to the defined folder structure.Can you please correct it?
```suggestion
                  import('@/pages/oidc-provider/add/AddOrUpdateOIDCProvider.page'),
```

File: packages/manager/apps/pci-kubernetes/src/routes.tsx [hunk 11]
```diff
@@ -125,13 +125,13 @@ export default [
...
                 ),
               },
               {
                 path: 'update-oidc-provider',
                 ...lazyRouteConfig(() =>
-                  import('@/pages/oidc-provider/UpdateOIDCProvider.page'),
+                  import('@/pages/oidc-provider/AddOrUpdateOIDCProvider.page'),
```
Comment (anooparveti) [inline packages/manager/apps/pci-kubernetes/src/routes.tsx:134]: **nitpick:** Each page/route must be a folder according to the defined folder structure.Can you please correct it?
```suggestion
                  import('@/pages/oidc-provider/update/AddOrUpdateOIDCProvider.page'),
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Review by SimonChaumet — DISMISSED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by SimonChaumet — DISMISSED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by SimonChaumet — DISMISSED** [review]

**Review by selm3n — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**